### PR TITLE
Fix networking issue

### PR DESCRIPTION
**Pull request by hydai**

### CODE REVIEW

**Comment by CaptainVincent** [comment]
According advice from docker team use [Multipass](https://github.com/canonical/multipass) test docker-desktop, but folow [install guide](https://docs.docker.com/desktop/install/ubuntu/) launch docker-desktop service not work. Setup verifies environment is not so easy as I think.

File: src/instance.rs [hunk 1]
```diff
@@ -182,11 +201,76 @@ impl Instance for Wasi {
...
+                        // Either way, this is an indication that the hook command
+                        // finished execution.  If the hook command was successful,
+                        // which we will check later in this function, we should not
+                        // fail this step here. We still want to check for all the other
+                        // error, in the case that the hook command is waiting for us to
+                        // write to stdin.
+                        let state = format!("{{ \"pid\": {} }}", std::process::id());
+                        let encoded_state =
+                            serde_json::to_string(&state).context("failed to encode container state")?;
+                        if let Err(e) = stdin.write_all(encoded_state.as_bytes()) {
```
Comment (rumpl) [inline src/instance.rs]: We actually don't need the encoded state here, the `state` is already a JSON string.
```suggestion
                        if let Err(e) = stdin.write_all(state.as_bytes()) {
```

Comment (hydai) [reply src/instance.rs]: Fixed. Thank you!

**Review by CaptainVincent — APPROVED** [review]
LGTM.

**Review by rumpl — APPROVED** [review]
👍

**Pull request merged**